### PR TITLE
Refactored codebase and Created data sources for other components

### PR DESCRIPTION
**Pull request by baimamboukar**

### CODE REVIEW

File: _dashboard/DotNetPerfMonitor/public/app_config.json [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+          "color": "rgb(255, 184, 76)"
+        },
+        {
+          "name": "Orleans-eda972a",
+          "color": "rgb(44, 211, 225)"
+        },
+        {
+          "name": "NuGetClient-win-d76a117",
+          "color": "rgb(255, 99, 132)"
+        }
```
Comment (marcin-krystianc) [inline _dashboard/DotNetPerfMonitor/public/app_config.json:44]: It is ok, for now - but think how to assign colours programmatically so we could add more benchmarks to the `nuget.csv` file without need to update the `app_coinfig.json` file.

File: .github/workflows/benchmarks.yml [hunk 2]
```diff
@@ -109,20 +109,23 @@ jobs:
...
-          Get-ChildItem -File -Path artifacts -Depth 1 | Foreach {. python3 process_results.py $_.fullname data.csv}
+          Get-ChildItem -File -Path artifacts -Depth 1 | Foreach {. python3 process_results.py $_.fullname ../data/nuget.csv}
 
       - name: Generate graph with Plotly for Python
+        working-directory: scripts/processors/
         if: ${{ false }}
         shell: pwsh
         run: |
-          . python3 generate_html.py data.csv index-python.html
+          . python3 generate_html.py ../data/nuget.csv index-python.html
```
Comment (marcin-krystianc) [inline .github/workflows/benchmarks.yml:122]: This step can be removed, it is a remnant from the old system.

File: .github/workflows/benchmarks.yml [hunk 3]
```diff
@@ -109,20 +109,23 @@ jobs:
...
 
       - name: Get alerts
+        working-directory: scripts/processors/
         id: alerts
         run: |
-          python3 generate_alert.py data.csv active_regressions.txt
+          python3 generate_alert.py ../data/nuget.csv active_regressions.txt
```
Comment (marcin-krystianc) [inline .github/workflows/benchmarks.yml:128]: Check also the other references to `data.csv`/`nuget.csv` or `active_regressions.txt` in the next steps. I think they need to be updated as well.

**Review by marcin-krystianc — COMMENTED** [review]
I would also rename the `benchmarks.yml` to `benchmarks_nuget.csv` and rename the name of the workflow in the line 1.

**Comment by baimamboukar** [comment]
> I would also rename the `benchmarks.yml` to `benchmarks_nuget.csv` and rename the name of the workflow in the line 1.

Okay! I can do it now....

File: _dashboard/DotNetPerfMonitor/public/app_config.json (hunk 1 above)
Comment (baimamboukar) [reply _dashboard/DotNetPerfMonitor/public/app_config.json:44]: Yeah! I created a composable `useColorGenerator` to randomly a RGB color.

But it is so dynamic, anytime the chart is loaded, the colors change.

That's why I added colors in app_config ......

If this is not OK I can simply switch back to color generator we used to have

File: .github/workflows/benchmarks.yml (hunk 2 above)
Comment (baimamboukar) [reply .github/workflows/benchmarks.yml:122]: Alright! ✅🚀

File: .github/workflows/benchmarks.yml (hunk 3 above)
Comment (baimamboukar) [reply .github/workflows/benchmarks.yml:128]: Okay!

**Review by marcin-krystianc — APPROVED** [review]
Looks like all necessary places were updated. Thanks.

**Review by marcin-krystianc — APPROVED** [review]

**Pull request merged**